craft_application/application.py

```diff
@@ -156,6 +156,9 @@ def __init__(
         else:
             self._work_dir = pathlib.Path.cwd()
 
+        # Whether the command execution should use the fetch-service
+        self._use_fetch_service = False
```

Comment:
I never caught this in previous work, but maybe this should go under `AppMetadata`?

Reply:
You mean as an `AppFeature`? The last agreement we had on this is that the support for the fetch-service integration will be enabled by default, and this variable controls whether the given run should use the fetch-service (that is, if the application was called with `--use-fetch-service`)